src/CabanaPD_Force.hpp

```diff
@@ -1026,38 +1026,36 @@ class Force<ExecutionSpace, ForceModel<PMB, Fracture>>
                                                                   i );
             for ( std::size_t n = 0; n < num_neighbors; n++ )
             {
-                if ( mu( i, n ) > 0 )
+                double fx_i = 0.0;
+                double fy_i = 0.0;
+                double fz_i = 0.0;
+
+                std::size_t j =
+                    Cabana::NeighborList<NeighListType>::getNeighbor(
+                        neigh_list, i, n );
+
+                // Get the reference positions and displacements.
+                double xi, r, s;
+                double rx, ry, rz;
+                getDistanceComponents( x, u, i, j, xi, r, s, rx, ry, rz );
+
+                // Break if beyond critical stretch unless in no-fail zone.
+                if ( r * r >= break_coeff * xi * xi && !nofail( i ) &&
+                     !nofail( j ) )
+                {
+                    mu( i, n ) = 0;
+                }
+                else if ( mu( i, n ) > 0 )
```

Comment:
Needs same comment as above

Reply:
Here, we do not have to ensure m(i) or m(j) are not 0.

Reply:
Then can't we remove this condition entirely?

Reply:
Removing it is probably fine. I haven't seen much difference in terms of performance on GPUs though.

Reply:
I'd say let's remove it for simplicity since it won't affect correctness

Reply:
I just checked and, actually, we do get improved performance with the else if statement: 

With "else if ( mu( i, n ) > 0 )":
-----------------------------
#Procs Particles | Time T_Force T_Comm T_Int T_Other T_Init |
8 11520000 | 316.99 294.74 3.92 0.08 15.18 3.07 | PERFORMANCE
8 11520000 | 1.00 0.93 0.01 0.00 0.05 0.01 | FRACTION

Without "else if ( mu( i, n ) > 0 )":
-----------------------------
#Procs Particles | Time T_Force T_Comm T_Int T_Other T_Init |
8 11520000 | 335.11 309.83 4.01 0.08 18.11 3.08 | PERFORMANCE
8 11520000 | 1.00 0.92 0.01 0.00 0.05 0.01 | FRACTION

So, the force calculation time reduces by ~4.8% and the communication time reduces by ~2.2%